assets/src/editor/events/tickets/editor-ticket/edit-form/new-ticket-form-modal.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * External imports
+ */
+import { useCallback, useEffect, useState } from '@wordpress/element';
+import { useOpenEditor } from '@eventespresso/editor-hocs';
+import {
+	useBasePriceType,
+	useCreateTicketEntity,
+} from '@eventespresso/hooks';
+
+/**
+ * Internal imports
+ */
+import EditTicketFormModal from './edit-ticket-form-modal';
+import useTicketEditorId from './use-ticket-editor-id';
+
+/**
+ * @param {Object} props
+ * @member {boolean} loading
+ * @member {Function} setLoading
+ * @return {Object} rendered Edit Ticket Form Modal
+ */
+const NewTicketFormModal = ( { loading, setLoading } ) => {
+	const [ newTicket, cacheNewTicket ] = useState( null );
+	const [ toggleTicketEditor, setToggleTicketEditor ] = useState( false );
+	const basePriceType = useBasePriceType();
+	const createTicketEntity = useCreateTicketEntity(
+		cacheNewTicket,
+		basePriceType
+	);
+	const editorId = useTicketEditorId( newTicket, 'new-ticket' );
+	const openTicketEditor = useOpenEditor( editorId );
+	// using the button click event to create the new ticket entity
+	// was causing issues with re-renders due to the async nature of things.
+	// triggering everything after render makes the UI run more smoothly
+	useEffect( () => {
+		if ( loading ) {
+			setLoading( false );
+			createTicketEntity();
+			setToggleTicketEditor( true );
+		}
+	} );
+	// because we have to wait for a valid ticket entity to be created,
+	// we can't simply open the editor via the Add New Ticket click event,
+	// so instead we toggle the following flag to indicate this
+	if ( toggleTicketEditor ) {
+		openTicketEditor();
+	}
+	// once the ticket editor has been opened, we can flip that toggle to off
+	const onEditorOpen = useCallback( () => {
```

Comment:
`useCallback` [returns a memoized callback](https://reactjs.org/docs/hooks-reference.html#usecallback).
This isn't an expensive function, we can simply have:
`const onEditorOpen = () => setToggleTicketEditor( false );`

Reply:
my understanding is that when you pass callbacks down as props that they can trigger rerenders because a new instance of the callback is generated each time and therefore any comparisons will fail, so even though it may not be an expensive function, it could cause unnecessary renders for a component that IS expensive. 

If I'm misinterpreting something here then I can change this.